XWebView/XWVHttpConnection.swift

```diff
@@ -0,0 +1,280 @@
+                    }
+                    bytesConsumed += data.length
+                }
+                ++ptr
+            }
+            if bytesConsumed > 0 {
+                // Move remained bytes to the begining.
+                inputBuffer.replaceBytesInRange(NSRange(0..<bytesConsumed), withBytes: nil, length: 0)
+            } else if bytesReaded + cursor == inputBuffer.length {
+                // Enlarge input buffer.
+                guard inputBuffer.length < bufferMaxSize else {
+                    close()
+                    break
+                }
+                inputBuffer.length <<= 1
+            }
+            cursor += bytesReaded - bytesConsumed
+            if output.hasSpaceAvailable { fallthrough }
+
+        case NSStreamEvent.HasSpaceAvailable:
+            if outputBuffer == nil {
+                guard let request = requestQueue.popLast() else { break }
+                let response = delegate.handleRequest(request)
+                if request.HTTPMethod == "GET", let fileURL = response.URL where fileURL.fileURL {
+                    fileHandle = try! NSFileHandle(forReadingFromURL: fileURL)
+                    fileSize = Int(fileHandle.seekToEndOfFile())
+                }
+                outputBuffer = response.octetsOfHeaders
+                bytesRemained = outputBuffer.length + fileSize
+            }
+
+            var bytesSent = 0
+            repeat {
+                let off: Int
+                if bytesRemained > fileSize {
+                    // Send response header
+                    off = outputBuffer.length - (bytesRemained - fileSize)
+                } else {
+                    // Send file content
+                    off = (fileSize - bytesRemained) % bufferMaxSize
+                    if off == 0 {
+                        fileHandle.seekToFileOffset(UInt64(fileSize - bytesRemained))
+                        outputBuffer = fileHandle.readDataOfLength(bufferMaxSize)
+                    }
+                }
+                let ptr = UnsafePointer<UInt8>(outputBuffer.bytes)
+                bytesSent = output.write(ptr.advancedBy(off), maxLength: outputBuffer.length - off)
+                bytesRemained -= bytesSent
+            } while bytesRemained > 0 && output.hasSpaceAvailable && bytesSent > 0
+            if bytesRemained == 0 {
+                // Response has been sent completely.
+                fileHandle = nil
+                fileSize = 0
+                outputBuffer = nil
+            }
+            if bytesSent < 0 { fallthrough }
+
+        case NSStreamEvent.ErrorOccurred:
+            print("<XWV> ERROR: " + (aStream.streamError?.localizedDescription ?? "Unknown"))
+            fallthrough
+
+        case NSStreamEvent.EndEncountered:
+            fileHandle = nil
+            inputBuffer = nil
+            outputBuffer = nil
+            close()
+
+        default:
+            break
+        }
+    }
+}
+
+private extension String {
```

Comment:
I think it's better to put this extension in a separated file like `String.swift`